model/src/form/form-metadata/index.ts

```diff
@@ -21,10 +21,18 @@ export const organisations = [
 
 export const idSchema = Joi.string().hex().length(24).required()
 export const titleSchema = Joi.string().max(250).trim().required()
-export const slugSchema = Joi.string().required()
+export const slugSchema = Joi.string()
```

Comment:
I don't think this processing belongs here on `slugSchema`.
If `slug` was a user inputted field, then this code is correct, but it isn't.
Slug is just a string, built from `title` using _some algorithm_ that lives in `forms-manager`. We should export that and use it here in `forms-designer` (or better still use [slugify](https://www.npmjs.com/package/slugify) or similar so we don't stumble over the probably many edge cases there are.)

Having the logic in two places feels off and open to us having a mismatch.

Also, `slugSchema` is re-used in other validation. E.g. in the `forms-runner`, if the user goes to `/my--slug`, we should take whatever `slug` param they send ("my--slug") and use it verbatim and not have joi turn it into "my-slug"

Reply:
That's fair, I've set up a new `slugify()` helper in:

* https://github.com/DEFRA/forms-designer/pull/149

I was going to import `slugSchema` into the other repos but this works too

Let me know what you think